src/IO/ADIOS/ADIOS2Auxiliary.cpp

```diff
@@ -23,6 +23,10 @@
 #if openPMD_HAVE_ADIOS2
 #include "openPMD/IO/ADIOS/ADIOS2Auxiliary.hpp"
 #include "openPMD/Datatype.hpp"
+#define OPENPMD_SWITCHTYPE_ADIOS2
```

Comment:
See above please, not a fan of such constructs because they are too error prune.
Also every macro controlling header files is an issue towards future C++20 module support.

Reply:
I guess I should read up at some point on how C++20 modules work exactly